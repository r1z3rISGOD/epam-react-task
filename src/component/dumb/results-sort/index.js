import {ResultsSort} from "./results-sort"
export {ResultsSort}
