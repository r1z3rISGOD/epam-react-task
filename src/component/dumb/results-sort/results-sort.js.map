{"version":3,"file":"results-sort.js","sourceRoot":"","sources":["results-sort.tsx"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;AAAA,6BAA8B;AAC9B,+BAA4B;AAC5B,kDAA2F;AAC3F,2CAAyC;AACzC,4DAAuD;AAUhD,IAAM,WAAW,GAAgC,UAAC,EAMxD;QALC,MAAM,YAAA,EACN,SAAS,eAAA,EACT,YAAY,kBAAA,EACZ,aAAa,mBAAA,EACb,UAAU,gBAAA;IAEV,IAAM,QAAQ,GAAG,yBAAW,EAAE,CAAA;IAE9B,IAAI,YAAY,GAAG,CAAC,EAAE;QACpB,IAAI,MAAM,KAAK,cAAc,EAAE;YAC7B,OAAO,CACG,6BAAK,SAAS,EAAC,cAAc;gBACzB,8BAAM,SAAS,EAAC,oBAAoB,cAAe;gBACnD,gCAAQ,KAAK,EAAC,cAAc,EAAC,SAAS,EAAC,8CAA8C,mBAAsB;gBAC3G,gCAAQ,OAAO,EAAG;;;;;oCAChB,SAAS,CAAC,cAAc,CAAC,CAAA;oCACzB,QAAQ,CAAC,oCAA0B,CAAC,cAAc,CAAC,CAAC,CAAA;oCACnC,qBAAM,wBAAS,CAAC,UAAU,EAAE,aAAa,EAAE,cAAc,CAAC,EAAA;;oCAArE,QAAQ,GAAG,SAA0D;oCAC3E,QAAQ,CAAC,gCAAsB,CAAC,QAAQ,CAAC,IAAI,CAAC,CAAC,CAAA;;;;yBAChD,EAAE,KAAK,EAAC,cAAc,EAAC,SAAS,EAAC,oBAAoB,aAAgB,CACpE,CACf,CAAA;SACF;QAED,IAAI,MAAM,KAAK,cAAc,EAAE;YAC7B,OAAO,CACG,6BAAK,SAAS,EAAC,cAAc;gBACzB,8BAAM,SAAS,EAAC,oBAAoB,cAAe;gBACnD,gCAAQ,OAAO,EAAE;;;;;oCACf,SAAS,CAAC,cAAc,CAAC,CAAA;oCACzB,QAAQ,CAAC,oCAA0B,CAAC,cAAc,CAAC,CAAC,CAAA;oCACnC,qBAAM,wBAAS,CAAC,UAAU,EAAE,aAAa,EAAE,cAAc,CAAC,EAAA;;oCAArE,QAAQ,GAAG,SAA0D;oCAC3E,QAAQ,CAAC,gCAAsB,CAAC,QAAQ,CAAC,IAAI,CAAC,CAAC,CAAA;;;;yBAChD,EAAE,KAAK,EAAC,cAAc,EAAC,SAAS,EAAC,oBAAoB,mBAAsB;gBAC5E,gCAAQ,KAAK,EAAC,cAAc,EAAC,SAAS,EAAC,8CAA8C,aAAgB,CACnG,CACf,CAAA;SACF;KACF;SAAM;QACL,OAAO,CACC,yCAAK,CACZ,CAAA;KACF;AACH,CAAC,CAAA;AA5CY,QAAA,WAAW,eA4CvB"}