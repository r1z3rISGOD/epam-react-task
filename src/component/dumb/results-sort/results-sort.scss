@import "./src/_variables/variables.scss";

.results-sort__text {
  font-size: 1.5rem;
  font-family: $font;
}

.results-sort__link {
  font-size: 1.5rem;
  font-family: $font;
  margin-left: 20px;
  text-decoration: none;
  color: black;
  background-color: transparent;
  border: none;
  cursor: pointer;
}

.results-sort__link-active {
  color: $color;
}
