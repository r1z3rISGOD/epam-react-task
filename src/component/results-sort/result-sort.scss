$color: #e55f64;
$font: "Roboto", sans-serif;

.results-sort__text {
  font-size: 1.5rem;
  font-family: $font;
}

.results-sort__link {
  font-size: 1.5rem;
  font-family: $font;
  margin-left: 30px;
  text-decoration: none;
  color: black;
}

.results-sort__link-active {
  color: $color;
}
