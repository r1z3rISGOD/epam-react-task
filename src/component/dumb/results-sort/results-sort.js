import React from 'react'
import './results-sort.scss'
import { changeTypeOfResultsSorting, putFetchedArrayToStore } from '../../../store/actions'
import { useDispatch } from 'react-redux'
import { fetchData } from '../../services/fetchService'

export const ResultsSort = (props) => {
  const dispatch = useDispatch()
  const { active, setActive, resultsCount, searchSetting, searchData } = props

  if (resultsCount > 0) {
    if (active === 'release_date') {
      return (
                <div className="results-sort">
                    <span className="results-sort__text">Sort by</span>
                    <button value="release_date" className="results-sort__link results-sort__link-active">release date</button>
                    <button onClick={ async () => {
                      setActive('vote_average')
                      dispatch(changeTypeOfResultsSorting('vote_average'))
                      const response = await fetchData(searchData, searchSetting, 'vote_average')
                      dispatch(putFetchedArrayToStore(response.data))
                    }} value="vote_average" className="results-sort__link">rating</button>
                </div>
      )
    }

    if (active === 'vote_average') {
      return (
                <div className="results-sort">
                    <span className="results-sort__text">Sort by</span>
                    <button onClick={async () => {
                      setActive('release_date')
                      dispatch(changeTypeOfResultsSorting('release_date'))
                      const response = await fetchData(searchData, searchSetting, 'release_date')
                      dispatch(putFetchedArrayToStore(response.data))
                    }} value="release_date" className="results-sort__link">release date</button>
                    <button value="vote_average" className="results-sort__link results-sort__link-active">rating</button>
                </div>
      )
    }
  } else {
    return (
            <></>
    )
  }
}
