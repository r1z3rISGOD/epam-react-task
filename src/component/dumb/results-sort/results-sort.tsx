import * as React from 'react'
import './results-sort.scss'

interface ResultsSortProps {
    searchSetting: string
    searchData: string
    sortingResults: any
    sortVision: number
}

export const ResultsSort : React.FC<ResultsSortProps> = ({
  searchSetting,
  searchData,
  sortingResults,
  sortVision
}) => {
  switch (sortVision) {
    case 1:
      return (
                <div className="results-sort">
                    <span className="results-sort__text">Sort by</span>
                    <button value="release_date" className="results-sort__link results-sort__link-active">release date</button>
                    <button onClick={async () => await sortingResults('vote-average', searchSetting, searchData)} value="vote_average" className="results-sort__link btn">rating</button>
                </div>
      )
    case 2:
      return (
                <div className="results-sort">
                    <span className="results-sort__text">Sort by</span>
                    <button onClick={async () => await sortingResults('release_date', searchSetting, searchData)} value="release_date" className="results-sort__link btn">release date</button>
                    <button value="vote_average" className="results-sort__link results-sort__link-active">rating</button>
                </div>
      )
    default:
      return (
              <></>
      )
  }
}
