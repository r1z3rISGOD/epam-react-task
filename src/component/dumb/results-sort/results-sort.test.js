import { ResultsSort } from './results-sort.tsx'
import React from 'react'

const mockSortingResults = jest.fn((value, setting, data) => { console.log(value, setting, data) })

const setUp = (sort) => shallow(<ResultsSort searchSetting='title' searchData='harry' sortingResults={mockSortingResults} sortVision={sort}/>)

describe('results sort test', () => {
  it('results sort test 1', () => {
    const comp = setUp(1)
    expect(comp).toMatchSnapshot()
    const btn = comp.find('.btn')
    btn.simulate('click')
    expect(mockSortingResults.mock.calls.length).toBe(1)
  })
  it('results sort test 2', () => {
    const comp = setUp(2)
    expect(comp).toMatchSnapshot()
    const btn = comp.find('.btn')
    btn.simulate('click')
    expect(mockSortingResults.mock.calls.length).toBe(1)
  })
  it('results sort test 3', () => {
    const comp = setUp(0)
    expect(comp).toMatchSnapshot()
  })
})
