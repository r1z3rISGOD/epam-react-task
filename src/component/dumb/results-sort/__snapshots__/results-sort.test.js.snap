// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`results sort test results sort test 1 1`] = `
<div
  className="results-sort"
>
  <span
    className="results-sort__text"
  >
    Sort by
  </span>
  <button
    className="results-sort__link results-sort__link-active"
    value="release_date"
  >
    release date
  </button>
  <button
    className="results-sort__link btn"
    onClick={[Function]}
    value="vote_average"
  >
    rating
  </button>
</div>
`;

exports[`results sort test results sort test 2 1`] = `
<div
  className="results-sort"
>
  <span
    className="results-sort__text"
  >
    Sort by
  </span>
  <button
    className="results-sort__link btn"
    onClick={[Function]}
    value="release_date"
  >
    release date
  </button>
  <button
    className="results-sort__link results-sort__link-active"
    value="vote_average"
  >
    rating
  </button>
</div>
`;

exports[`results sort test results sort test 3 1`] = `<Fragment />`;
